addons/medical_statemachine/functions/fnc_localityChangedEH.sqf

```diff
@@ -19,15 +20,47 @@
 params ["_unit", "_isLocal"];
 TRACE_2("localityChangedEH",_unit,_isLocal);
 
-if (!alive _unit) exitWith {TRACE_1("dead", _this)};
-
-if (!_isLocal) then {
-    // If locality changed, broadcast the last medical state
-    _unit setVariable [VAR_HEART_RATE, GET_HEART_RATE(_unit), true];
-    _unit setVariable [VAR_BLOOD_PRESS, _unit getVariable [VAR_BLOOD_PRESS, [80, 120]], true];
-    _unit setVariable [VAR_BLOOD_VOL, GET_BLOOD_VOLUME(_unit), true];
+if (!alive _unit) exitWith {};
 
+if (_isLocal) then {
     private _currentState = [_unit, EGVAR(medical,STATE_MACHINE)] call CBA_statemachine_fnc_getCurrentState;
-    TRACE_2("sending current state",_unit,_currentState);
-    [QGVAR(localityTransfer), [_unit, _currentState], _unit] call CBA_fnc_targetEvent;
+    TRACE_1("local",_currentState);
+
+    switch (true) do {
+        case (IN_CRDC_ARRST(_unit)): {
+            if (_currentState == "CardiacArrest") exitWith {};
+            _unit setVariable [VAR_CRDC_ARRST, false]; // force reset vars so setCardiacArrest can run (enteredStateCardiacArrest will also be called)
+            _unit setVariable [VAR_UNCON, false];
+            TRACE_1("manually changing state to CardiacArrest",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "CardiacArrest", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        case (IS_UNCONSCIOUS(_unit)): {
+            if (_currentState == "Unconscious") exitWith {};
+            _unit setVariable [VAR_UNCON, false]; // force reset var so ace_medical_status_fnc_setUnconscious can run
+            TRACE_1("manually changing state to Unconscious",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Unconscious", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        // case (Injured?) - probably not needed, "Default" and "Injured" are the same
```

Comment:
Should be fine, only thing would be the [FullHeal](https://github.com/acemod/ACE3/blob/master/addons/medical_statemachine/Statemachine.hpp#L30) event. The unit should go into Injured (from Default) if they are [bleeding or in pain](https://github.com/acemod/ACE3/blob/master/addons/medical_vitals/functions/fnc_handleUnitVitals.sqf#L152), but there could be a situation where neither is the case?

Reply:
Added injured check and now does transition